### DISCUSSION TITLE
Missing cppadcg.hpp header file

### DISCUSSION THREAD

**Discussion by CharlesJQuarra** [coin-or/CppAD#122] [Q&A] [answered]
Hi,

This header includes a header that I couldn't find in the repository:

https://github.com/coin-or/CppAD/blob/master/include/cppad/example/code_gen_fun.hpp#L15

which is also referenced on the core library (https://github.com/coin-or/CppAD/blob/d66c1e4cebeec7dd9f9fa291e14385915ec9cd9d/cppad_lib/code_gen_fun.cpp), so as it is it would probably fail to build, am I missing something?

**Comment by bradbell** [comment, marked as answer]
If you look at
https://coin-or.github.io/CppAD/doc/cmake.htm#include_cppadcg
you will see the text:

include_cppadcg
The cppadcg examples will be compiled and tested if include_cppadcg=true is on the command line. Warning : Do not use this option when installing cppad because the cppadcg package depends on cppad and using this option makes cppad depend on cppadcg. This option, and the script get_cppadcg.sh are only intended for testing purposes. 

See
https://github.com/joaoleal/CppADCodeGen

You can get around this by first install cppad with include_cppadcg false.
Then install CppADCodeGen.
Then install cppad with include_cppadcg true.
This is what is done to make the tests of CppAD not depend on CppADCodegen.

A better solution is to add the example utility
https://coin-or.github.io/CppAD/doc/code_gen_fun.htm
to the CppADCodeGen library using its source code  and documentation in CppAD as an example of how to do this.

**Comment by bradbell** [comment]
@CharlesJQuarra Did the message above answer your question ?

**Comment by CharlesJQuarra** [comment]
yes it did, thanks!

**Comment by bradbell** [comment]
This was moved here from issue#122 on 2021-10-27.
